pyro/distributions/log_normal.py

```diff
@@ -32,24 +32,11 @@ def __init__(self, mu, sigma, *args, **kwargs):
             raise ValueError("Expected mu.size() == sigma.size(), but got {} vs {}".format(mu.size(), sigma.size()))
         super(LogNormal, self).__init__(*args, **kwargs)
 
-    def batch_shape(self, x=None):
-        event_dim = 1
-        mu = self.mu
-        if x is not None:
-            if x.size()[-event_dim] != mu.size()[-event_dim]:
-                raise ValueError("The event size for the data and distribution parameters must match.\n"
-                                 "Expected x.size()[-1] == self.mu.size()[-1], but got {} vs {}".format(
-                                     x.size(-1), mu.size(-1)))
-            try:
-                mu = self.mu.expand_as(x)
-            except RuntimeError as e:
-                raise ValueError("Parameter `mu` with shape {} is not broadcastable to "
-                                 "the data shape {}. \nError: {}".format(mu.size(), x.size(), str(e)))
-        return mu.size()[:-event_dim]
+    def batch_shape(self):
+        return self.mu.size()
 
     def event_shape(self):
-        event_dim = 1
-        return self.mu.size()[-event_dim:]
+        return ()
```

Comment:
It's probably a little safer to return `torch.Size(())` here, since `torch.Tensor(())` treats its argument as an iterable, whereas `torch.Tensor(torch.Size(()))` treats its argument as a shape.

Reply:
Will change.